TopNews/app/src/main/java/com/fisfam/topnews/MainActivity.java

```diff
@@ -89,6 +96,35 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    //Choose country of news source
+    private void chooseCountry() {
+        mUserPref = new UserPreference(getApplicationContext());
```

Comment:
fine but no need, "this" for activity context is also fine. Since the context is "short lived" - meaning it only survive until the method is complete. 
If in other case you want to store the context in an instance variable for instance, then you should use either WeakReference or at least ApplicationContext

Reply:
su khac nhau giua this, getContext(), getActivity(), and maybe some more la gi? Doi khi hoi mong lung ko biet chon cai nao

Reply:
again: https://stackoverflow.com/questions/32227146/what-is-different-between-getcontext-and-getactivity-from-fragment-in-support-li/32227256

Reply:
Da doc cai tren. Thay van con vai cai methods nua de get context. Still confused :D